src/main/java/uk/gov/hmcts/dts/fact/services/admin/AdminCourtOpeningTimeService.java

```diff
@@ -0,0 +1,75 @@
+package uk.gov.hmcts.dts.fact.services.admin;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.dts.fact.entity.Court;
+import uk.gov.hmcts.dts.fact.entity.CourtOpeningTime;
+import uk.gov.hmcts.dts.fact.exception.NotFoundException;
+import uk.gov.hmcts.dts.fact.model.OpeningTime;
+import uk.gov.hmcts.dts.fact.repositories.CourtRepository;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static java.util.stream.Collectors.toList;
+
+@Service
+public class AdminCourtOpeningTimeService {
+    private final CourtRepository courtRepository;
+
+    @Autowired
+    public AdminCourtOpeningTimeService(final CourtRepository courtRepository) {
+        this.courtRepository = courtRepository;
+    }
+
+    public List<OpeningTime> getCourtOpeningTimesBySlug(final String slug) {
+        return courtRepository.findBySlug(slug)
+            .map(c -> c.getCourtOpeningTimes()
+                .stream()
+                .map(CourtOpeningTime::getOpeningTime)
+                .map(OpeningTime::new)
+                .collect(toList()))
+            .orElseThrow(() -> new NotFoundException(slug));
+    }
+
+    public List<OpeningTime> updateCourtOpeningTimes(final String slug, final List<OpeningTime> openingTimes) {
+        final Court courtEntity = courtRepository.findBySlug(slug)
+            .orElseThrow(() -> new NotFoundException(slug));
+        return saveNewOpeningTimes(courtEntity, openingTimes);
+    }
+
+    private List<OpeningTime> saveNewOpeningTimes(final Court courtEntity, final List<OpeningTime> openingTimes) {
+        List<uk.gov.hmcts.dts.fact.entity.OpeningTime> openingTimesEntity = getNewOpeningTimesEntity(openingTimes);
+        List<CourtOpeningTime> courtOpeningTimesEntity = getNewCourtOpeningTimesEntity(courtEntity, openingTimesEntity);
+
+        if (courtEntity.getCourtOpeningTimes() == null) {
```

Comment:
Food for thought if we have Guava, we can use:

if (Strings.isNullOrEmpty(<string>)) which checks if it's both null, and also if not, whether or not the string within is empty or not

Reply:
getCourtOpeningTimes() returns a list of objects not strings. We could use Apache common or Spring framework CollectionUtils.isEmpty() in this case. I can change it